Pipeline: output: loki: style (#1719)
fluent/fluent-bit-docs: pipeline/outputs/loki.md

}
```

Add the JSON path to the plugin output configuration:

```python
[OUTPUT]
    name   loki
    match  *
    label_map_path /path/to/map.json
```

The previous configurations accomplish the same as this one:

```python
[OUTPUT]
    name   loki
    match  *
    labels job=fluentbit, $sub['stream']
```

Both will generate the following Streams label:

```text
job="fluentbit", stream="stdout"
```

#### Kubernetes and labels

> If you are running in a Kubernetes environment, consider enabling the `auto_kubernetes_labels` option, which auto-populates the streams with the Pod labels for you. Consider the following configuration:

Reviewer: ```suggestion
If you're running in a Kubernetes environment, consider enabling the `auto_kubernetes_labels` option, which auto-populates the streams with the Pod labels for you. Consider the following configuration:
```

    auto_kubernetes_labels on
```

Based on the JSON example provided, the internal stream labels will be:

```text
job="fluentbit", team="Santiago Wanderers"
```

## Use `drop_single_key`

> If there is only one key remaining after removing keys, you can use the `drop_single_key` property to send its value to Loki, rather than a single `key=value` pair.

Reviewer: ```suggestion
If there's only one key remaining after removing keys, you can use the `drop_single_key` property to send its value to Loki, rather than a single `key=value` pair.
```

value
```

> If you want both structured JSON and plain text logs in Loki, you should set `drop_single_key` to `raw` and `line_format` to `json`.

Reviewer: ```suggestion
If you want both structured JSON and plain text logs in Loki, set `drop_single_key` to `raw` and `line_format` to `json`.
```

Structured metadata is officially supported starting with Loki 3.0, and shouldn't be used
with Loki deployments prior to Loki 3.0.

### Structured metadata maps

In addition to the `structured_metadata` configuration parameter, a `structured_metadata_map_keys` is available, which can be used to dynamically populate structured metadata from map values in the log record. `structured_metadata_map_keys` can be set with a list of record accessors, where each one should reference map values in the log record. Record accessors which don't match a map value will be skipped.

> The following configuration is similar to the previous example, except now all entries in the log record map value `$kubernetes` will be used as structured metadata entries.

Reviewer: ```suggestion
The following configuration is similar to the previous example, except now all entries in the log record map value `$kubernetes` will be used as structured metadata entries:
```

    labels                       cluster=my-k8s-cluster, region=us-east-1
    structured_metadata_map_keys $kubernetes
```

{% endtab %}

{% tab title="fluent-bit.yaml" %}

```yaml
  outputs:
    - name:                         loki
      match:                        *
      labels:                       cluster=my-k8s-cluster, region=us-east-1
      structured_metadata_map_keys: $kubernetes
```

{% endtab %}
{% endtabs %}

Assuming the value `$kubernetes` is a map containing two entries `namespace_name` and `pod_name`, the previous configuration is equivalent to:

{% tabs %}
{% tab title="fluent-bit.conf" %}

```python
[OUTPUT]
    name                loki
    match               *
    labels              cluster=my-k8s-cluster, region=us-east-1
    structured_metadata $kubernetes['namespace_name'],$kubernetes['pod_name']
```

{% endtab %}

{% tab title="fluent-bit.yaml" %}

```yaml
  outputs:
    - name:                loki
      match:               *
      labels:              cluster=my-k8s-cluster, region=us-east-1
      structured_metadata: $kubernetes['namespace_name'], $kubernetes['pod_name']
```

{% endtab %}
{% endtabs %}

## Networking and TLS Configuration

> This plugin inherits core Fluent Bit features to customize the network behavior and optionally enable TLS in the communication channel. For more details about the specific options available refer to the following articles:

Reviewer: ```suggestion
This plugin inherits core Fluent Bit features to customize the network behavior and optionally enable TLS in the communication channel. For more details about the specific options available, refer to the following articles:
```

- [Networking Setup](../../administration/networking.md): timeouts, keepalive and source address
- [Security and TLS](../../administration/transport-security.md): all about TLS configuration and certificates

All options mentioned in these articles must be enabled in the plugin configuration in question.

### Fluent Bit and Grafana Cloud

Fluent Bit supports sending logs and metrics to [Grafana Cloud](https://grafana.com/products/cloud/) by providing the appropriate URL and ensuring TLS is enabled.

> An example configuration, set the credentials and ensure the host URL matches the correct one for your deployment:

Reviewer: ```suggestion
Aa an example configuration, set the credentials and ensure the host URL matches the correct one for your deployment:
```
